drivers/storage/port/scsiport/scsi.c

```diff
@@ -683,8 +682,23 @@ SpiProcessCompletedRequest(
     /* Flush adapter if needed */
     if (SrbInfo->BaseOfMapRegister)
     {
-        /* TODO: Implement */
-        ASSERT(FALSE);
+        if (SrbInfo->ScatterGather)
+        {
+            ULONG transferLen = 0;
+            BOOLEAN isWrite = Srb->SrbFlags & SRB_FLAGS_DATA_OUT ? TRUE : FALSE;
+
+            for(int i = 0; i < SrbInfo->NumberOfMapRegisters; i++)
+            {
+                transferLen += SrbInfo->ScatterGather[i].Length;
```

Comment:
I see you removed the check of `transferLen` against `Srb->DataTransferLength`. Is that not needed after all?

Reply:
Opps, Fixed